cmake/bitcode_rules.cmake

```diff
@@ -127,7 +127,7 @@ function(compile_sleef_c_to_bc EXT FILENAME SUBDIR BCLIST)
         "-I" "${CMAKE_SOURCE_DIR}/lib/kernel/sleef/arch"
         "-I" "${CMAKE_SOURCE_DIR}/lib/kernel/sleef/libm"
         "-I" "${CMAKE_SOURCE_DIR}/lib/kernel/sleef/include"
-        "-xc" "-o" "${BC_FILE}" "-c" "${FULL_F_PATH}"
+        "-O1" "-xc" "-o" "${BC_FILE}" "-c" "${FULL_F_PATH}"
```

Comment:
IMO this is a Clang bug or at least questionable behavior. Please check if this fixed upstream. It's common to run opts for the .bc produced by Clang with the 'opt' tool, it's weird that Clang would restrict the inlining that can be applied with opt. I think it hasn't been like this in the older versions of Clang.

Reply:
Looking at LLVM trunk, it's still there:

``./tools/clang/lib/Frontend/CompilerInvocation.cpp``  around line 454:

```
  // At O0 we want to fully disable inlining outside of cases marked with
  // 'alwaysinline' that are required for correctness.
  Opts.setInlining((Opts.OptimizationLevel == 0)
                       ? CodeGenOptions::OnlyAlwaysInlining
                       : CodeGenOptions::NormalInlining);
```

Git blame:

```
commit d0ca1639669b4ad12088c1b4d84312b7091f1666
Author: Chandler Carruth <chandlerc@gmail.com>
Date:   Fri Dec 23 01:24:49 2016 +0000

    Cleanup the handling of noinline function attributes, -fno-inline,
    -fno-inline-functions, -O0, and optnone.
```

So you're right that this hasn't been the case with older versions of Clang. We could probably workaround this somewhere in pocl-llvm-API but the -O1 seems to be the simplest solution.